model/src/main/proto/exploration.proto

```diff
@@ -253,6 +253,13 @@ message UserAnswer {
   string content_description = 5;
 }
 
+message HintState {
```

Comment:
Perhaps ``StatePlayerSavedHintState`` might be a bit clearer on what this contains & how it's used. Unfortunately it uses two meanings of the same word (state) but I had trouble coming up with an alternative that didn't & was equally clear.

Please also add documentation comments for the message & each field here.

Reply:
Done added it the new file `ui_state.proto`